### ISSUE TITLE
shipit_static_analysis: *.pb.cc and *.pb.h files should be ignored

### ISSUE THREAD

**Issue by fmarier** [mozilla/release-services#699]
**MozReview URL:** https://reviewboard.mozilla.org/r/195238/diff/1#index_header

**Problem:** These files are generated by the protocol buffer compiler. They should be ignored.

**Comment by abpostelnicu** [comment]
I see your point, but if we exclude them how are we going to know if they generate issues? My take on this is that we should keep them in the analysis and for each reviewer/committer should also modify the scrips that generate these files.

**Comment by sylvestre** [comment]
As Andi said, from my experience, running SA on generated code has some value, it can help finding issues in the code generator!

**Comment by fmarier** [comment]
You both make good points. I guess it's good to know that the protobuf compiler is generating less-than-ideal code.

It sort of creates the expectation that someone would have to fix the protobuf compiler to emit more modern C++, but I guess as long as developers can make the call that it's too much work (and probably won't be merged upstream until Google updates their internal version to more modern C++) then it's probably fine to leave them in.

Maybe the message that the bot posts on there should clarify the expectations?

e.g. "If the following defects are in automatically-generated code, then please consider fixing the generator or filing a bug upstream if appropriate."

**Comment by jankeromnes** [comment]
Thanks for reporting this issue!

I agree that it's valuable to be aware of bugs in generated code, especially if we can fix them in the generator, but I think we should also take into account the volume of such defects being reported vs how easy it is to actually fix them.

I think it's similar to defects found in third-party code, or in expanded macros (we currently silence both of these): They are genuine code defects that we will ship with our product (which is bad), but if the problem is small enough, or if it's too much work to fix it, then we shouldn't spam developers with lots of unhelpful defect reports.

TL;DR If clangbot reports many defects in generated code, but the generators are third-party code, or they're too hard to fix by the developers who are using them, then maybe we should consider silencing these reports. (However, if they don't happen that often, and fixing generators is usually easy, then we should definitely keep them).

**Comment by La0** [comment]
Outdated with the move to Try.
